From 6477b1ee9535f8df68db21322d5e8dc0c8077291 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 25 May 2016 13:12:46 -0700
Subject: [PATCH] afpd: don't use network IDs without LDAP

We can't get valid network IDs without a valid LDAP configuration. So
automatically set the NONETIDS volume flag when not using LDAP.

This fixes ACL setting errors (-50) related to getnamefromuuid() when
connected from a client logged in with a username and UID matching the
logged-in account on the server.

Signed-off-by: Justin Maggard <jmaggard@netgear.com>
---
 etc/afpd/acls.h   | 3 ---
 etc/afpd/volume.c | 3 ++-
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/etc/afpd/acls.h b/etc/afpd/acls.h
index 4d423af..3066405 100644
--- a/etc/afpd/acls.h
+++ b/etc/afpd/acls.h
@@ -112,9 +112,6 @@ int afp_access (AFPObj *obj, char *ibuf, size_t ibuflen, char *rbuf,  size_t *rb
 int afp_getacl (AFPObj *obj, char *ibuf, size_t ibuflen, char *rbuf,  size_t *rbuflen);
 int afp_setacl (AFPObj *obj, char *ibuf, size_t ibuflen, char *rbuf,  size_t *rbuflen);
 
-/* Parse afp.conf */
-extern int acl_ldap_readconfig(char *name);
-
 /* Misc funcs */
 extern int acltoownermode(const AFPObj *obj, const struct vol *vol, char *path, struct stat *st, struct maccess *ma);
 #endif
diff --git a/etc/afpd/volume.c b/etc/afpd/volume.c
index a707236..8665d80 100644
--- a/etc/afpd/volume.c
+++ b/etc/afpd/volume.c
@@ -40,6 +40,7 @@
 #include <atalk/unix.h>
 #include <atalk/netatalk_conf.h>
 #include <atalk/server_ipc.h>
+#include <atalk/ldapconfig.h>
 
 #ifdef CNID_DB
 #include <atalk/cnid.h>
@@ -379,7 +380,7 @@ static int getvolparams(const AFPObj *obj, uint16_t bitmap, struct vol *vol, str
                         ashort |= VOLPBIT_ATTR_UNIXPRIV;
                     if (vol->v_flags & AFPVOL_TM)
                         ashort |= VOLPBIT_ATTR_TM;
-                    if (vol->v_flags & AFPVOL_NONETIDS)
+                    if (!ldap_config_valid || vol->v_flags & AFPVOL_NONETIDS)
                         ashort |= VOLPBIT_ATTR_NONETIDS;
                     if (obj->afp_version >= 32) {
                         if (vol->v_vfs_ea)
-- 
2.8.3

